### PR TITLE
Truly disconnect Dashboard from the database

### PR DESCRIPTION
**Pull request by iszulcdeepsense**
User account management has been moved away from Dashboard to Lifecycle.
Dashboard no longer speaks directly to a database.
It should also improve Dashboard's responsiveness as Postgres is no longer its hard dependency, thus it doesn't hang indefinitely in case of a database malfunction.

### CODE REVIEW

**Review by Koshmaar — APPROVED** [review]
LGTM.

File: dashboard/dashboard/view/account.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        context['user_auth'] = get_auth_token(request)
+    except Exception as e:
+        log_exception(ContextError('Getting user profile data failed', e))
+        context['error'] = str(e)
+
+    return render(request, 'racetrack/profile.html', context)
+
+
+@login_required
+def retrieve_user_token(request):
```
[the remark this answers is not in the document]
Comment (iszulcdeepsense) [reply dashboard/dashboard/view/account.py:159]: This is rather a simple API endpoint returning data in HTTP response, as opposed to views returning full HTML pages. That's why it's not named `view_...`. I could name it `endpoint_...`, but views are actually endpoints too with a specific content type.

File: dashboard/dashboard/view/account.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+                'error': str(root_exception),
+                'register_username': username,
+            })
+
+    return render(request, 'registration/register.html', {
+        'register_username': '',
+    })
+
+
+def view_password_reset(request):
```
[the remark this answers is not in the document]
Comment (iszulcdeepsense) [reply dashboard/dashboard/view/account.py:111]: Indeed, `view_change_password` should require being logged in. I'll fix that. However, it doesn't apply to `view_password_reset`. Usually, you want to reset a password, when you're not logged in :wink:

File: racetrack_commons/racetrack_commons/entities/users_client.py [hunk 3]
```diff
@@ -1,24 +1,29 @@
...
+class UserAccountClient:
+    def __init__(self, auth_token: str = ''):
         self.lc_client = LifecycleClient(auth_token)
 
-    def get_user_profile(self, username: str) -> UserProfileDto:
-        username_encoded = quote_plus(username)
-        response = self.lc_client.request_dict('get', f'/api/v1/users/{username_encoded}/profile')
+    def register_user(self, username: str, password: str):
+        self.lc_client.request('post', f'/api/v1/users/register',
+                               json={'username': username, 'password': password})
```
[the remark this answers is not in the document]
Comment (iszulcdeepsense) [reply racetrack_commons/racetrack_commons/entities/users_client.py:12]: Yes, it raises an error on failure

File: racetrack_commons/racetrack_commons/entities/users_client.py [hunk 4]
```diff
@@ -1,24 +1,29 @@
...
-    def init_user_profile(self, username: str) -> UserProfileDto:
-        username_encoded = quote_plus(username)
-        response = self.lc_client.request_dict('post', f'/api/v1/users/{username_encoded}/profile')
+    def validate_token(self) -> UserProfileDto:
+        response = self.lc_client.request_dict('get', f'/api/v1/users/validate_user_auth')
         return parse_dict_datamodel(response, UserProfileDto)
 
+    def change_password(self, old_password: str, new_password: str):
+        self.lc_client.request('put', f'/api/v1/users/change_password',
+                               json={'old_password': old_password, 'new_password': new_password})
```
[the remark this answers is not in the document]
Comment (iszulcdeepsense) [reply racetrack_commons/racetrack_commons/entities/users_client.py:25]: Same here. It raises exception on failure, eg. 401 response

File: racetrack_commons/racetrack_commons/entities/users_client.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Koshmaar) [reply racetrack_commons/racetrack_commons/entities/users_client.py:12]: Fair enough. AFAIK, in python it's recommended to list in comment exceptions that are thrown by function. Maybe doing that in every function is excessive, but it seems here this rule fits, as the API differs from other functions.

File: dashboard/dashboard/view/account.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Koshmaar) [reply dashboard/dashboard/view/account.py:82]: Haha, my bad about the `view_password_reset`.

File: dashboard/dashboard/view/account.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Koshmaar) [reply dashboard/dashboard/view/account.py:131]: Ok.

File: racetrack_commons/racetrack_commons/entities/users_client.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (iszulcdeepsense) [reply racetrack_commons/racetrack_commons/entities/users_client.py:12]: Fine, I'll add a docstring.

**Pull request merged**